mdpow/workflows/dihedrals.py

```diff
@@ -544,30 +614,65 @@ def plot_violins(df, resname, figdir=None, molname=None, width=0.9, solvents=SOL
     if molname is None:
         molname = resname
 
+        # figdir is now necessary??
     if figdir is not None:
         subdir = molname
         newdir = os.path.join(figdir, subdir)
         os.mkdir(newdir)
 
-    section = df.groupby(by='selection')
+    section = df.groupby(by="selection")
 
+    # conversion factor: 1 mm = 3.7795275591 px
+    plot_pdf_width = plot_pdf_width * 3.7795275591
+    # DEFAULT: 190 mm = 718.110236229 pixels
+
+    # put this into separate function
     for name in section:
+        a = list(ab_pairs[name[0]][0])
+        b = list(ab_pairs[name[0]][1])
+
+        drawer = rdMolDraw2D.MolDraw2DSVG(250, 250)
+        drawer.DrawMolecule(mol=mol, highlightAtoms=a, highlightBonds=b)
+        drawer.FinishDrawing()
+        svg = drawer.GetDrawingText().replace('svg:','')
+
+        mol_svg = sg.fromstring(svg)
+        m = mol_svg.getroot()
+        m.scale(0.0125).moveto(21.8, 0.35)
+
         plot = dihedral_violins(name[1], width=width, solvents=solvents)
-        plot.set_titles(f'{molname},{name[0]}, ''{col_name}')
+        _ = plot.set_titles(f'{molname}, {name[0]} {list(ab_pairs[name[0]][0])} | ''{col_name}')
         # plot.set_titles needs to stay here during future development
         # this locale ensures that plots are properly named,
-        # especially when generated for a projecct iteratively
+        # especially when generated for a project iteratively
+
+        plot_svg = sg.from_mpl(plot)
+        p = plot_svg.getroot()
+        p.scale(0.02)
+
+        mol_svg = sg.fromstring(svg)
+        m = mol_svg.getroot()
+        m.scale(0.0125).moveto(21.8, 0.35)
+
+        # 28cm leaves room for lengthier solvent names
+        # in the legend on the rightmost side
+        # width and height for SVG render, previously 30, 5
+        # the order matters here, plot down first, mol on top
+        fig = sc.Figure("28cm", "4.2cm", p, m)
 
-        if figdir is not None:
-            figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
-            plot.savefig(figfile)
-            logger.info(f'Figure saved as {figfile}')
+        # figdir is now necessary and plots are not returned with molecule graphic
+        figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
+        plot_pdf = cairosvg.svg2pdf(bytestring=fig.tostr(), write_to=str(figfile),
+                                    output_width=plot_pdf_width)
 
-    return plot
+        logger.info(f"Figure saved as {figfile}")
+
+    return logger.info(f"All figures generated and saved in {figdir}")
```

Comment:
Returning the logger.info() does not make a lot of sense. 

```suggestion
    logger.info(f"All figures generated and saved in {figdir}")
    return
```

Right now, return `None` but then we'll update to return a df for all the created artifacts.